src/profile-logic/marker-data.js

```diff
@@ -598,22 +605,30 @@ export function deriveMarkersFromRawMarkerTable(
       }
 
       case 'IPC': {
-        const times = ipcCorrelations.get(threadId, i);
+        const pairData = ipcCorrelations.get(threadId, i);
+        let otherThreadName;
+        if (pairData) {
+          otherThreadName = `${friendlyThreadNames.get(
```

Comment:
I think this could be computed in the Tooltip view layer instead of here. This way we don't have to create a bunch of strings that may never be viewed.

https://github.com/firefox-devtools/profiler/blob/master/src/components/tooltip/Marker.js#L232

I think this component could also just compute the friendly thread name directly using the threads and current thread, rather than using the selector `getFriendlyThreadNames`.

Reply:
This is a bit more complex since I wouldn't be passing the "threads and current thread", but rather the threads and *target* thread. That's doable (probably), but it means looking up the target thread by its TID and then calling `ProfileData.getFriendlyThreadName`, once per IPC marker.

I'm concerned that this would end up being considerably more work than just making a "TID -> friendly name" mapping one time and caching it as a selector. I also wasn't sure how to pipe the extra data I needed to generate a friendly name all the way to the view functions, especially for `getMarkerFullDescription`, which only takes a marker as input.

Maybe it would make sense to take the existing `getFriendlyThreadName` selector and have it call my `getFriendlyThreadNames` selector though? I'm not 100% sure on what the perf difference would be there...

Reply:
> I'm concerned that this would end up being considerably more work than just making a "TID -> friendly name" mapping one time and caching it as a selector

Ah, I didn't realize the complexity here! This is fine then. Could you add a comment to the selector explaining this?

If you feel that this is the simpler implementation, then I'm fine moving forward. I just assumed the tooltip component would have access to all of the information it needed.